### PR TITLE
feature/SDK-13 getting list of keys

### PR DESCRIPTION
**Pull request by sksadjad**

### CODE REVIEW

File: packages/kms-local/src/SphereonKeyManagementSystem.ts [hunk 1]
```diff
@@ -268,4 +268,8 @@ export class SphereonKeyManagementSystem extends KeyManagementSystem {
     const signer = new RSASigner(PEMToJwk(hexToPEM(publicKeyHex, 'public'), 'public'), { hashAlgorithm, scheme })
     return await signer.verify(data, signature)
   }
+
+  public async listKeys(): Promise<ManagedKeyInfo[]> {
+    return (await this.privateKeyStore.list({})).map((privateKey) => this.asSphereonManagedKeyInfo(privateKey))
```
Comment (nklomp) [inline packages/kms-local/src/SphereonKeyManagementSystem.ts:273]: Never ever use the private key store. We store both public and private keys. The private keys store is only used for signing and encryption. Never us that for external facing methods!

**Comment by nklomp** [comment]
The KeyStore has a listKeys. If I am not mistaken this does already do everything you need (get the keys from all KMS implementations registered)

File: packages/kms-local/src/SphereonKeyManagementSystem.ts (hunk 1 above)
Comment (sksadjad) [reply packages/kms-local/src/SphereonKeyManagementSystem.ts:273]: Initially, I found it unusual to work with the private key store for this. But the `asSphereonManagedKeyInfo` does the trick and strips the private key part from this

Comment (sksadjad) [reply packages/kms-local/src/SphereonKeyManagementSystem.ts:273]: pushed a commit that fails if the returned keys have the `privateKeyHex` property

**Pull request merged**